### PR TITLE
Create GitHub releases during production rollouts

### PR DESCRIPTION
**Pull request by dkurepa**
https://github.com/dotnet/arcade-services/issues/3551

### Release Note Category
- [ ] Feature changes/additions 
- [ ] Bug fixes
- [x] Internal Infrastructure Improvements

### Release Note Description
Create GitHub releases during production rollouts

### CODE REVIEW

File: eng/templates/stages/deploy.yaml [hunk 1]
```diff
@@ -104,6 +105,12 @@ stages:
         PublishProfilePath: $(Pipeline.Workspace)/MaestroApplication/projectartifacts/PublishProfiles/${{ parameters.PublishProfile }}.xml
         ApplicationPackage: $(Pipeline.Workspace)/MaestroApplication/applicationpackage
 
+    - ${{ if eq(variables['Build.SourceBranch'], 'refs/heads/production')}}:
+      - powershell: gh release create "v$(releaseVersion)" --generate-notes --latest --title "Rollout $(currentDate)" --target $(Build.SourceVersion) --repo dotnet/arcade-services
+        displayName: Create Release
```
Comment (premun) [inline eng/templates/stages/deploy.yaml:110]:
```suggestion
        displayName: Create GitHub release
```

**Review by premun — APPROVED** [review]
Nice

**Review by tkapin — COMMENTED** [review]
One question to think about - This tagging scheme accommodates one release per day. What do we do if we need to do multiple releases? Wouldn't it be better to use the pipeline build number, that uniquely identifies the build?

File: eng/templates/stages/deploy.yaml [hunk 2]
```diff
@@ -104,6 +105,12 @@ stages:
         PublishProfilePath: $(Pipeline.Workspace)/MaestroApplication/projectartifacts/PublishProfiles/${{ parameters.PublishProfile }}.xml
         ApplicationPackage: $(Pipeline.Workspace)/MaestroApplication/applicationpackage
 
+    - ${{ if eq(variables['Build.SourceBranch'], 'refs/heads/production')}}:
+      - powershell: gh release create "v$(releaseVersion)" --generate-notes --latest --title "Rollout $(currentDate)" --target $(Build.SourceVersion) --repo dotnet/arcade-services
```
Comment (premun) [inline eng/templates/stages/deploy.yaml:109]:
```suggestion
      - powershell: gh release create "v$(releaseVersion)" --generate-notes --latest --title "Rollout $(currentDate) / $(Build.BuildNumber)" --target $(Build.SourceVersion) --repo dotnet/arcade-services
```

**Review by premun — DISMISSED** [review]

File: eng/templates/stages/deploy.yaml [hunk 3]
```diff
@@ -104,6 +105,22 @@ stages:
...
+        artifact: AssetManifests
+
+      - powershell: |
+          $xmlPaths = Get-ChildItem $(Pipeline.Workspace)/AssetManifests/ -Filter *.xml
+          $xml = [xml](Get-Content $xmlPaths[0].FullName)
+          $version = $xml.Build.Package[0].Version
+          Write-Host "##vso[task.setvariable variable=releaseVersion]$version"
+        displayName: Get Release Version
+
+      - powershell: gh release create "v$(releaseVersion)" --generate-notes --latest --title "Rollout $(currentDate) / $(Build.BuildNumber)" --target $(Build.SourceVersion) --repo dotnet/arcade-services
```
Comment (premun) [inline eng/templates/stages/deploy.yaml:119]: One last thing please

```suggestion
      - powershell: gh release create "v$(Build.BuildNumber)-$(releaseVersion)" --generate-notes --latest --title "Rollout $(currentDate) / $(Build.BuildNumber)" --target $(Build.SourceVersion) --repo dotnet/arcade-services
```

**Review by premun — DISMISSED** [review]

**Review by premun — DISMISSED** [review]

**Review by premun — APPROVED** [review]

**Pull request merged**